.github/workflows/functional-test-noncloud.yaml

```diff
@@ -372,9 +374,13 @@ jobs:
       - name: Install Flux Source Controller
         if: matrix.name == 'kubernetes-noncloud'
         run: |
-          curl -s https://fluxcd.io/install.sh | FLUX_VERSION=2.5.1 sudo bash
-          flux install --namespace=flux-system --version=v2.5.1 --components=source-controller --network-policy=false
-          kubectl wait --for=condition=available deployment -l app.kubernetes.io/component=source-controller -n flux-system --timeout=120s
+          for i in 1 2 3; do
```

Comment:
How about making this an action or a shell script so that the code is not repeated in two places?